services/streammanager/lambda/lambdamanager.go

```diff
@@ -0,0 +1,100 @@
+//go:generate mockgen --build_flags=--mod=mod -destination=../../../mocks/services/streammanager/lambda/mock_lambda.go -package mock_lambda github.com/rudderlabs/rudder-server/services/streammanager/lambda LambdaClient
+
+package lambda
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/aws/aws-sdk-go/service/lambda"
+	"github.com/rudderlabs/rudder-server/services/streammanager/common"
+	"github.com/rudderlabs/rudder-server/utils/awsutils"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+	"github.com/tidwall/gjson"
+)
+
+// Config is the config that is required to send data to Lambda
+type Config struct {
+	InvocationType string
+	ClientContext  string
+	Lambda         string
+}
+
+type LambdaProducer struct {
+	client LambdaClient
+}
+
+type LambdaClient interface {
+	Invoke(input *lambda.InvokeInput) (*lambda.InvokeOutput, error)
+}
+
+var pkgLogger logger.LoggerI
+
+func init() {
+	pkgLogger = logger.NewLogger().Child("streammanager").Child(lambda.ServiceName)
+}
+
+// NewProducer creates a producer based on destination config
+func NewProducer(destinationConfig map[string]interface{}, o common.Opts) (*LambdaProducer, error) {
+	sessionConfig, err := awsutils.NewSessionConfig(destinationConfig, o.Timeout, lambda.ServiceName)
+	if err != nil {
+		return nil, err
+	}
+	if sessionConfig.Region == "" {
+		return nil, errors.New("could not find region configuration")
+	}
+	awsSession, err := awsutils.CreateSession(sessionConfig)
+	if err != nil {
+		return nil, err
+	}
+	return &LambdaProducer{client: lambda.New(awsSession)}, nil
+}
+
+// Produce creates a producer and send data to Lambda.
+func (producer *LambdaProducer) Produce(jsonData json.RawMessage, _ interface{}) (int, string, string) {
+	client := producer.client
+	if client == nil {
+		return simpleErrorResponse("Could not create client")
+	}
+	data := gjson.GetBytes(jsonData, "payload").String()
+	if data == "" {
+		return simpleErrorResponse("Invalid payload")
+	}
+	destConfig := gjson.GetBytes(jsonData, "destConfig").String()
+	if destConfig == "" {
+		return simpleErrorResponse("Invalid Destination Config")
+	}
+
+	var config Config
+	err := json.Unmarshal([]byte(destConfig), &config)
+	if err != nil {
+		return simpleErrorResponse("error while unmarshalling destination config :: " + err.Error())
+	}
```

Comment:
`[non-blocking]` I have a few issues with the usage of `gjson`:
1. Performance: Every time we use GetBytes, we have to scan a significant portion of the byte array.
2. Readability: It is hard to understand the structure of the expected payload, you have to scan the code to find all the expected properties. In this case, is not that bad because we only have two properties in the same area, but in other situations, it can become really ugly.

I have a suggestion that maintains the same behaviour. 

NOTE: I am using pointers to check whether `destConfig` is set or not 

Assuming:
```go
type inputEvent struct {
         Payload string `json:"payload"`
         Config *Config  `json:"destConfig"`
}
```

```suggestion
	err := json.Unmarshal([]byte(destConfig), &event)
	if err != nil {
		return simpleErrorResponse("error while unmarshalling :: " + err.Error())
	}
	
	if event.Payload == "" {
			return simpleErrorResponse("Invalid payload")
	}
	if event.Config == nil {
		return simpleErrorResponse("Invalid Destination Config")
	}
```